File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program121.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which checks whether 15th bit is On or OFF.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

typedef unsigned int UINT;

class Solution
{
	private:

		UINT iNo;
		UINT iMask;
		UINT iResult;

	public:
		Solution(UINT iNo)
		{
			this->iNo = iNo;
			this->iMask = 0x00004000;
                        this->iResult = 0;

		}

		bool CheckBit()
		{	
			iResult = iNo & iMask;
	
			if(iResult==iMask)
			{
				return true;
			}
			else
			{
				return false;
			}
		}
};	

int main()
{
	UINT iValue=0;
	bool bRet=false;

	cout<<"Enter number: ";
	cin>>iValue;

	Solution *sObj = new Solution(iValue);
	bRet = sObj->CheckBit();

	if(bRet == true)
	{
		cout<<"15th bit is ON\n";
	}
	else
	{
		cout<<"15th bit is OFF\n";
	}
}

/*
OUTPUT:

Enter number: 13
15th bit is OFF

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program121.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which checks whether 15th bit is On or OFF.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	public int iMask,iNo;
	public int iResult;

	Solution(int iNo)
	{
		this.iNo = iNo;
		this.iMask = 0x00004000;
		this.iResult = 0;
	}

	boolean CheckBit()
	{

		iResult = iNo & iMask;

		if(iResult==iMask)
		{
			return true;
		}
		else
		{
			return false;
		}
	}	
}

class Program
{
	public static void main(String[] args)
	{
		boolean bRet=false;

		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number: ");
		int iValue = sObj.nextInt(); 

		Solution sObj1 = new Solution(iValue);
		bRet = sObj1.CheckBit();

		if(bRet == true)
		{
			
			System.out.println("15th bit is ON");
		}
		else
		{
			System.out.println("15th bit is OFF");
		}
	}	
}
/*
OUTPUT:

Enter number: 
16384
15th bit is ON

*/
















